client/client.go

```diff
@@ -88,6 +91,8 @@ type Client struct {
 	// so that subsequent requests do not re-negotiate.
 	negotiateVersion bool
 
+	// this mutex protects negotiated field.
+	negotiatedMu sync.RWMutex
```

Comment:
Slightly thinking if we need separate locking for this one, or if we should use a single one, as I think they would be set "in tandem".

There's also `manualOverride`; would that need a lock as well?

Reply:
I removed this mutex and used `versionMu` for both fields instead.

Regarding fields like `manualOverride` or `negotiateVersion`: they are only set using `Opt` functions during client configuration, before it is used. Therefore they cannot be read concurrently by client users. When the client is ready, these fields are never changed, so they don't need protection.